lib/utils.nix

```
@@ -125,27 +125,95 @@ rec {
     end
   '';
 
+  /**
+    Convert the given String to a Lua [long literal].
+
+    [long literal]: https://www.lua.org/manual/5.4/manual.html#3.1
+
+    # Examples
+
+    ```nix
+    nix-repl> toLuaLongLiteral "simple"
+    "[[simple]]"
+    ```
+
+    ```nix
+    nix-repl> toLuaLongLiteral "]]"
+    "[=[]]]=]"
+    ```
+
+    # Type
+
+    ```
+    toLuaLongLiteral :: String -> String
+    ```
+  */
+  toLuaLongLiteral =
+    string:
+    let
+      buildTokens =
+        depth:
+        let
+          infix = lib.strings.replicate depth "=";
+          tokens.open = "[${infix}[";
+          tokens.close = "]${infix}]";
+        in
+        if lib.hasInfix tokens.close string then buildTokens (depth + 1) else tokens;
+
+      tokens = buildTokens 0;
+    in
+    tokens.open + string + tokens.close;
+
+  /**
+    Conver the given String into a Vimscript [:let-heredoc], for use by [:lua].
```

```suggestion
    Convert the given string into a Vimscript [:let-heredoc].

    For example, this could be used to pass a lua string to the [:lua] command.
```

1. Typo `conver` -> `convert`
2. I don't _think_ "String" needs to be capitalized?
3. `:lua` isn't the only use case, so we should re-phrase it as a "for instance".

---

1. Typo fixed
2. Ah, thanks for pointing that out. This was intentional, to try evoke [this type hint I saw higher up in the file](https://github.com/jfly/nixvim/blob/40df98e4691f36cbc0a178aed20b10895183d22a/lib/utils.nix#L55). But perhaps it just looks weird, and isn't The Nix Way. This PR is probably the first time I've ever written a nix docstring.
3. Done.

---

Yeah it seems types used in the type-signature are usually capitalised. Outside that, in the prose of the docstring, I think we can be less formal.

But I'm fine either way if it was intended. :+1: